### ISSUE TITLE
Alternative styles buttons: insufficient focus indication

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#15906]
Indication of focus can't rely on color alone. It needs an additional shape.

Noticed while testing #10128 and it applies to all the Styles "buttons" (actually, they're not real `<button>` elements). 

To reproduce, use the keyboard and move through the Styles buttons. Examples from the Button component styles in #10128

Initial active style state: dark grey box-shadow

<img width="309" alt="Screenshot 2019-05-30 at 08 53 06" src="https://user-images.githubusercontent.com/1682452/58618827-d7119c80-82c3-11e9-978b-93dc747f4f49.png">

Focus on the active style: blue box-shadow: in absence of color, this can't be distinguished from the normal state:

<img width="297" alt="Screenshot 2019-05-30 at 08 53 18" src="https://user-images.githubusercontent.com/1682452/58618905-fa3c4c00-82c3-11e9-8bf3-1488ec996501.png">

Move focus to the adjacent button: blue box-shadow: in absence of color, this can't be distinguished from the normal state:

<img width="302" alt="Screenshot 2019-05-30 at 08 53 21" src="https://user-images.githubusercontent.com/1682452/58618947-0d4f1c00-82c4-11e9-801c-7c6617f9f89c.png">

Same applies to Styles for other components, e.g. the Quote:

<img width="308" alt="Screenshot 2019-05-30 at 09 50 56" src="https://user-images.githubusercontent.com/1682452/58619002-31aaf880-82c4-11e9-8c47-778f6928eb3b.png">

**Comment by kjellr** [comment]
These buttons use our standard block library styles, so there's no precedent for an active/selected state, combined with a focus state.

A few possible options: 

1. Add a second blue border when the active style is focused: 

_Located outside the box:_ 
![opt-1](https://user-images.githubusercontent.com/1202812/59309780-5060a600-8c59-11e9-87d8-f81676b1bbcc.gif)

_Or, located inside the box:_ 
![opt-3](https://user-images.githubusercontent.com/1202812/59309798-62424900-8c59-11e9-8c79-7f904ae14061.gif)

---

2. Implement a slightly thicker blue border on focus. 

_3px focus border:_ 
![opt-2](https://user-images.githubusercontent.com/1202812/59309825-771edc80-8c59-11e9-9b1e-2f565c052844.gif)

_4px focus border:_
![opt-5](https://user-images.githubusercontent.com/1202812/59309963-d11fa200-8c59-11e9-8b10-fcc9b0dc3500.gif)

---

3. Rather than alter the focus state, change the treatment of the active state:
![opt-6](https://user-images.githubusercontent.com/1202812/59310157-52773480-8c5a-11e9-98ff-e4e020f8f2a1.gif)

---

My preference is #3, so that we can continue using the same focus styles as the block library. For the treatment there, I've adopted the usual hover state to indicate that the item is active, and implemented a slightly darker background on hover.

**Comment by melchoyce** [comment]
I like 3 as well, but what if it also had a 1px grey border?

![image](https://user-images.githubusercontent.com/2846578/59310474-56578680-8c5b-11e9-96f9-7b7fe2ccc075.png)

**Comment by mapk** [comment]
I like number 3 too. It shows an active state similar to how icons show a background color when they are active in the block toolbar. Albeit they show a much darker background color, but it would probably be too much in this case.

![Screen Shot 2019-06-11 at 3 20 17 PM](https://user-images.githubusercontent.com/617986/59310814-72a7f300-8c5c-11e9-8a8b-382d6c4db569.png)

**Comment by kjellr** [comment]
Thanks, I agree. I tired that out initially, but it felt very heavy in context: 

<img width="280" alt="Screen Shot 2019-06-11 at 3 23 32 PM" src="https://user-images.githubusercontent.com/1202812/59311012-f3ff8580-8c5c-11e9-9566-3619792945a8.png">
<img width="345" alt="Screen Shot 2019-06-11 at 3 23 43 PM" src="https://user-images.githubusercontent.com/1202812/59311013-f3ff8580-8c5c-11e9-98dc-40892c59f0a1.png">

I'll spin up a PR with the lighter gray focus state. 👍

**Comment by kjellr** [comment]
Another option for the selected state would be to have some sort of check mark indicator, like we do for the colors panel options: 

<img width="183" alt="Screen Shot 2019-06-11 at 4 17 24 PM" src="https://user-images.githubusercontent.com/1202812/59313088-6a53b600-8c64-11e9-8e55-36fd7de0c716.png">

Maybe something like this: 

![checkmark](https://user-images.githubusercontent.com/1202812/59313376-928fe480-8c65-11e9-8b61-1d1d7b9ccaf6.png)

**Comment by kjellr** [comment]
Before I actually bring this into a PR, I wanted to check out how this may end up looking from a component basis. Here's the current state, with today's styles: 

<img width="591" alt="Frame 6 5" src="https://user-images.githubusercontent.com/1202812/59392632-2e385800-8d2d-11e9-92d4-7c63cdf4dfe1.png">

As noted above, this doesn't work for us because when the focus ring is applied to a "selected" state, there's just a single color state to indicate focus, which isn't adequate for accessibility. 

Here's a range of the possible options for the selected state: 

**Dark gray (mimics our "selected" state for iconButtons)**
<img width="591" alt="Frame 6 4" src="https://user-images.githubusercontent.com/1202812/59392768-b7e82580-8d2d-11e9-9fa3-5edcac609cbd.png">

**Blue**
<img width="591" alt="Frame 6 4" src="https://user-images.githubusercontent.com/1202812/59392848-0695bf80-8d2e-11e9-9861-79c8f439a5ab.png">

**Light gray**
<img width="591" alt="Frame 6 3" src="https://user-images.githubusercontent.com/1202812/59392795-d5b58a80-8d2d-11e9-8686-63ff065ff7b8.png">

**Light gray with check:** 
<img width="591" alt="Frame 6 2" src="https://user-images.githubusercontent.com/1202812/59392814-e49c3d00-8d2d-11e9-8342-d5e0f9083fa4.png">

**Comment by melchoyce** [comment]
Looking at all the options laid out like this, I think I have to reverse my previous opinion — I think the solid colors are most obviously selected.

**Comment by kjellr** [comment]
Yeah I agree. I'm leaning more towards the dark gray version than I was initially. The blue looks good visually, but I worry that it reads color-wise like a primary button, which isn't ideal.

**Comment by melchoyce** [comment]
That's a good point. I still find the grey really heavy, but maybe heavy is still preferable to subtle.

**Comment by afercia** [comment]
Just a quick consideration: the new focus style for input fields and textareas use a thin "outline" for the normal state. On focus, the outline changes color _and_ gets thicker. That's OK, as the thickness change is perceivable also without colors. Just wondering it a similar pattern could be used for the selected / focus states discussed here.

**Comment by kjellr** [comment]
> On focus, the outline changes color and gets thicker. That's OK, as the thickness change is perceivable also without colors. Just wondering it a similar pattern could be used for the selected / focus states discussed here.

Yeah — I didn't mock this up there but my thought was that we could adopt the outer ring style we use in the Button block for the focus states here: 

<img width="272" alt="Screen Shot 2019-06-13 at 7 19 40 AM" src="https://user-images.githubusercontent.com/1202812/59440328-a2194580-8dab-11e9-94fe-60c9550ecdc7.png">

This approach gets us a brand new outer border, which adds thickness and color to the focused item.

**Comment by espman** [comment]
Hi guys,
I just wanted to throw in these 2 ideas I sketched at the contributor day.
I know they are not thought through yet, but wanted to manifest these ideas here :) 

**Var 01a**
I wanted to show the idea of an overlapping checkmark to stand out more to emphasise the selected state.In this sketch it will touch the neighboured element, because of the small margin

so I had the Idea of **Var 01b**

**Var 02**
Or a complete different approach could be elevation levels (shadows) for hover, focus or selective state, which I think is also part of MD.

![github](https://user-images.githubusercontent.com/20412454/59963999-ed91c900-94fa-11e9-8640-374c26da182a.jpg)

**Comment by mapk** [comment]
Thanks for those additional thoughts, @espman! It's great to see the explorations with Material's elevation.

**Comment by kjellr** [comment]
I dove into this a bit more in the context of #16283, and I'm wondering if we're actually overthinking this a bit. Maybe the default + active borders just need to be inset, and the outside borders need to be outset: 

<img width="506" alt="Frame" src="https://user-images.githubusercontent.com/1202812/60602160-437b3200-9d81-11e9-82d7-ef4967ec513b.png">

This doubles up the borders on focus, but I think that feels totally fine.
